File: todolist/src/main/java/de/maeddes/todolist/TodolistApplication.java
package de.maeddes.todolist;

import java.util.ArrayList;
import java.util.List;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;

@SpringBootApplication
public class TodolistApplication {

	List<String> todos = new ArrayList<String>();

	@GetMapping("/todos/")
	List<String> getTodos() {

		return todos;
	}

	@PostMapping("/todos/{todo}")
	String addTodo(@PathVariable String todo) {

		todos.add(todo);
		return "added " + todo;
	}

	@DeleteMapping("/todos/{todo}")
	String removeTodo(@PathVariable String todo) {

		todos.remove(todo);
		return "removed " + todo;

	}

	public static void main(String[] args) {
		SpringApplication.run(TodolistApplication.class, args);
	}

}

File: todolistkt/src/main/kotlin/com/example/todolist/TodolistApplication.kt
package com.example.todolist

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.data.repository.CrudRepository
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import org.springframework.web.bind.annotation.*
import javax.persistence.*


@SpringBootApplication
@RestController
class TodolistApplication(val todoRepository: TodoRepository){

	@PostMapping("/todos/{todo}")
	fun addTodo(@PathVariable todo:String):String {
		todoRepository.save(Todo(todo))
		return "added " + todo
	}

	@GetMapping("/todos/")
	fun getTodos():List<String>{
		var todos = ArrayList<String>()
		todoRepository.findAll().forEach<Todo> { element -> todos.add(element.todo) }
		return todos

	}

	@DeleteMapping("/todos/{todo}")
	fun removeTodo(@PathVariable todo:String):String {
		todoRepository.deleteById(todo)
		return "removed " + todo
	}


}

fun main(args: Array<String>) {
	runApplication<TodolistApplication>(*args)
}

@Entity
data class Todo(@Id var todo: String = "") {
}

@RepositoryRestResource(path = "todo-hal")
interface TodoRepository: CrudRepository<Todo, String>